pep-0677.rst

```diff
@@ -850,6 +955,36 @@ from the Steering Council on ensuring that type expressions do not
 diverge from the rest of Python's syntax.
 
 
+Improving Usability of the Indexed Callable Type
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If we do not want to add new syntax for callable types, we could
+look at how to make the existing type easier to read. One proposal
+would be to make the builtin ``callable`` function indexable so
+that it could be used as a type::
+
+    callable[[int, str], bool]
+
+This change would be analogous to PEP 585 that made built in collections
+like ``list`` and ``dict`` usable as types, and would make imports
+more convenient, but it wouldn't help readability of the types themselves
+much.
+
+In order to reduce the number of brackets needed in complex callable
+types, it would be possible to allow tuples for the argument list::
+
+    callable[(int, str), bool]
+
+This actually is a significant readability improvement for multi-argument
+functions, but the problem is that it makes callables with no arguments, which
+are *very* common in typeshed, hard to write. Because ``()`` is not a legal
```

Comment:
That's not true
```
In [6]: len(())
Out[6]: 0

In [7]: type(())
Out[7]: tuple
```

Reply:
Oh oops, I hadn't had my coffee yet.

That's supposed to be talking about one-argument functions!